firefox-ios/Client/Frontend/Browser/Tabs/Middleware/TabManagerMiddleware.swift

```diff
@@ -447,8 +447,8 @@ class TabManagerMiddleware {
     /// Handles undo close all inactive tabs. Adding back the backup tabs saved previously
     private func undoCloseAllInactiveTabs(uuid: WindowUUID) {
         let tabManager = tabManager(for: uuid)
-        ensureMainThread {
-            tabManager.undoCloseInactiveTabs()
+        Task {
+            await tabManager.undoCloseInactiveTabs()
```

Comment:
At a quick glance this seems like a significant change (going from MT only to a detached `Task`). Was this added solely to clear up compiler warnings? Or was there a separate issue involved?

Reply:
The `undoCloseInactiveTabs()` func still runs on MT. If you got to the definition you can see that it has `@MainActor` attribute set.